source/vibe/data/serialization.d

```diff
@@ -446,8 +446,12 @@ private void serializeImpl(Serializer, alias Policy, T, ATTRIBUTES...)(ref Seria
 	} else static assert(false, "Unsupported serialization type: " ~ T.stringof);
 }
 
+private T deserializeImpl(T, alias Policy, Serializer, ATTRIBUTES...)(ref Serializer deserializer) if(!isMutable!T)
+{
+	return cast(T) deserializeImpl!(Unqual!T, Policy, Serializer, ATTRIBUTES)(deserializer);
```

Comment:
Wut. I always assumed it worked with everything and was just documenting.
I'm not familiar with this part, but could you also add an unittest for a `const`/`immutable` object if there's not one already ?